src/test/groovy/pl/tomaszdziurko/jvm_bloggers/blog_posts/domain/BlogPostSpec.groovy

```diff
@@ -0,0 +1,52 @@
+package pl.tomaszdziurko.jvm_bloggers.blog_posts.domain
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class BlogPostSpec extends Specification {
+
+    @Unroll
+    def "Should return \"#expectedState\" state when approved is #approved "(final Boolean approved, final String expectedState) {
+        given:
+            final BlogPost blogPost = new BlogPost(approved: approved)
+        when:
+            final String approvalState = blogPost.getApprovalState()
+        then:
+            approvalState == expectedState
+        where:
+            approved | expectedState
+            true     | "Approved"
+            false    | "Rejected"
+            null     | " -- "
+    }
+
+    @Unroll
+    def "Should return #expected when isModerated called for post with approved = #approved"(final Boolean approved, final boolean expected) {
+        given:
+            final BlogPost blogPost = new BlogPost(approved: approved)
+        when:
+            final boolean isModerated = blogPost.isModerated()
+        then:
+            isModerated == expected
+        where:
+            approved | expected
+            true     | true
+            false    | true
+            null     | false
+    }
+
+    @Unroll
+    def "Should return #expected when isNotModerated called for post with approved = #approved"(final Boolean approved, final boolean expected) {
+        given:
+            final BlogPost  blogPost = new BlogPost(approved: approved)
```

Comment:
I totally agree when it comes to production code. We all know that using "final" keyword there improves reasoning about the code. It is even more important, because production methods are likely to grow (that's why we refactor after some time). On the other hand, test methods have always much smaller flow than production ones and using "final" is a subject for discussion, as @tdziurko wrote. I agree with him using "final", "final" and "final" everywhere in tests make too much noise and there is too little gain from it.